### ISSUE TITLE
[TF2] (Linux) Common recurring freeze

### ISSUE THREAD

**Issue by Skincanvas** [ValveSoftware/Source-1-Games#4742]
I cannot reproduce it accurately, but my game freezes between 5 and 120 minutes, reliably.
This only happens in Team Fortress 2, and has only been a problem for the past 3 or so weeks.
Typically it freezes after I die, during the freezecam or immediately following a respawn, but several times just walking around or whatever.
And, unfortunately I cannot find any information that seems useful in any logs I could find so I am just sending everything I can thats somewhat relevant whether I think its useful or not.

[Indepth system information](https://gist.github.com/Skincanvas/a62b255b5f54b4f3f1921fa5c4c72176)
Basic system information;
- Nvidia RTX 2060 Super
- AMD Ryzen 5 3600X
- Arch Linux 64bit
- Linux with Zen patches, currently 6.1.7
- ext4 filesystem
- Xfce4
- Xorg

Here is all the information I have so far.
[gdb/terminal output](https://gist.github.com/Skincanvas/64c0cdd0d3d6fd5908993058aeb59875)
I end the game through the Steam library page which works but doesn't seem to output anything useful. Also I cut out literally 10000+ xlib spam messages. That spam can be 'fixed' by rolling back lib32-libx11 to 1.8.2-2. But that isn't for this issue, doing so doesn't fix the freezes anyways and seems to introduce crashes instead.

[gdb/terminal output with reverted libx11](https://gist.github.com/Skincanvas/b0b8b98f4851ba3209763dc15515762b)
This might be a different problem altogether, but in case its useful since I don't have anything else really, this is after a crash that is a lot more informative. I feel like it could be relevant, because until recently TF2 didn't even crash for me so perhaps this is the problem getting 'further'.

The following don't seem to correlate with the freezes. They do not update during a freeze so probably just some irrelevant shadercache messages but I am scraping now so here goes.
[dmesg and journalctl](https://gist.github.com/Skincanvas/e8fce6a4995ade6b392bb880ffd6c985)

If anyone has any ideas of where I could maybe find more information, I will gladly try to get it but these are the ones I know how to find.

The things I have tried;
1. Fresh Team Fortress 2 install, with no launch options.
2. Steam stable (currently on beta channel)
3. Checked hardware sensors, made sure they weren't like overheating or something but everything looked normal.
4. Changed from linux-zen to linux
5. Rolled back linux-zen+linux-zen-headers to version 6.1.1
6. Reinstalled Arch Linux entirely. Not usually my solution but I figured it would be a nice refreshment and a catchall

**Comment by kisak-valve** [comment]
Hello @Skincanvas, there are multiple known regressions with libX11 1.8.2 and 1.8.3 which need to be resolved in the upstream package. Please test how the game behaves with lib32-libx11 1.8.1 as well.

**Comment by Skincanvas** [comment]
Thank you for the very fast response, I didn't expect it. I forgot to save the terminal output unfortunately but I have just played for a little over an hour with lib32-libx11 1.8.1 + a reboot and it was working fine. However unless this is just a known issue I cannot yet state the problem has been fixed since there have been times I was playing successfully for over an hour before. I will continue testing this tomorrow evening for a longer period of time, for I have to get ready for work soon. Hopefully this is all I had to do, and if so I will update the title so people can find the thread better
